Dockerfile

```diff
@@ -0,0 +1,84 @@
+# syntax=docker/dockerfile:1.2
+#########################################################
+# BUILD IMAGE
+#########################################################
+
+# hadolint ignore=DL3006,DL3049
+FROM golang:1.17 AS build
+WORKDIR /app
+COPY go.mod go.sum ./
+RUN --mount=type=secret,id=netrc,dst=/root/.netrc \
+    --mount=type=cache,target=/go/src,sharing=locked \
+    --mount=type=cache,target=/go/pkg,sharing=locked \
+    --mount=type=cache,target=/root/.cache/go-build,sharing=locked \
+    go mod download
+COPY . ./
+RUN --mount=type=cache,target=/go/src,sharing=locked \
+    --mount=type=cache,target=/go/pkg,sharing=locked \
+    --mount=type=cache,target=/root/.cache/go-build,sharing=locked \
+    go build ./cmd/syncer
+
+#########################################################
+# TEST IMAGE
+# Rebuild the Go binary as source files change.
+#########################################################
+
+# hadolint ignore=DL3049
+FROM build AS test
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+# Install golangci-lint.
+RUN { curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "$(go env GOPATH)/bin" v1.41.1; } \
+    && ln -s /go/bin/golangci-lint /usr/local/bin/golangci-lint
+
+# hadolint ignore=DL3008
+RUN --mount=type=cache,target=/var/cache/apt \
+    --mount=type=cache,target=/var/lib/apt \
+    apt-get update -y && apt-get install -y --no-install-recommends \
+        jq \
+        rsync
+
+# Make the module cache a perminent part of the image.
+# hadolint ignore=DL3006,DL3018,DL3019
+RUN --mount=type=cache,id=/go/src,target=/gotmp/src,sharing=locked \
+    --mount=type=cache,id=/go/pkg,target=/gotmp/pkg,sharing=locked \
+    rsync -ra /gotmp/ /go/
+
+RUN go install github.com/githubnemo/CompileDaemon@v1.3.0
+CMD ["CompileDaemon", "-include", "*.go", "-include", "*.html", "-include", "go.*", "--command=./cmd/syncer"]
+
+#########################################################
+# RELEASE IMAGE
+#########################################################
+
+# hadolint ignore=DL3006,DL3007
+FROM ubuntu:latest AS release
+
+USER root
+
+COPY --from=build /app/syncer /app/syncer
+# ENTRYPOINT will not work because we use a bind mount (the volume does not exist until the container is running).
+# Use something like this when running this image:
+# docker run --mount type=bind,source=<DASHBOARD_JSON_DIRECTORY>,target=/app/dashboards --network="host" <THIS_IMAGE> /app/syncer -user admin -pass admin -dir /app/dashboards
```

Comment:
I'm not sure I understand why an ENTRYPOINT can't be used here.

Does this not work?
```Dockerfile
VOLUME ["/dashboards"]
ENTRYPOINT ["/app/syncer", "-dir", "/dashboards"]
# Use something like this when running this image:
# docker run --mount type=bind,source=<DASHBOARD_JSON_DIRECTORY>,target=/dashboards --network="host" <THIS_IMAGE> -user admin -pass admin
```

Reply:
Updated to use ENTRYPOINT ["/app/syncer"] in the latest revision. Users will still need to specify -user -pass and -dir for it to work